### DISCUSSION TITLE
Why is the basePath in Next Auth v5 still not working?

### DISCUSSION THREAD

**Discussion by hscstudio** [nextauthjs/next-auth#11169] [Help]
I have encountered several issues when implementing basePath, but unfortunately, the core auth developers only respond by sending irrelevant example links.

If basePath does indeed work, try setting it to something like basePath /custom, and I can confirm it won't work properly.

In my case, if the issue lies here, authentication succeeds but the redirect always goes to /api/auth/callback/keycloak?... instead of /custom/api/auth/callback/keycloak?....

If I manually change it to 'custom', then it works fine.

Just try it yourself if you don't believe me.

**Comment by jezza1245** [comment]
I am seeing the same thing #11321  . Is there an open issue for this? I have been trying to find a workaround but am struggling to do so; perhaps theres some rewrite or middleware trick that could be implemented; have you had any luck since?

**Comment by nktnet1** [comment]
There's an open pull request summarising the issue here (contains 2 workarounds):
- https://github.com/nextauthjs/next-auth/pull/10797

More discussions here:
- https://github.com/nextauthjs/next-auth/discussions/9274

Another proposed workaround:
- https://github.com/nextauthjs/next-auth/discussions/12160

**Comment by SrayasPMaatrum** [comment]
for v5 [/next-auth/discussions/9274](https://github.com/nextauthjs/next-auth/discussions/9274 )this works for me
